File: backend/schemas/graph.py
from pydantic import BaseModel
from typing import List

class Node(BaseModel):
    id: str
    type: str
    label: str
    x: int
    y: int

class Edge(BaseModel):
    source: str
    target: str

class Graph(BaseModel):
    nodes: List[Node]
    edges: List[Edge]

File: frontend/src/types/graph.ts
export const NODE_WIDTH = 180;
export const NODE_HEIGHT = 72;

export interface GraphNode {
  id: string;
  type: "service" | "database" | "queue" | "cache" | "client" | "generic";
  label: string;
  x: number;
  y: number;
}

export interface GraphEdge {
  id: string;
  source: string;
  target: string;
  label?: string;
}

export interface Graph {
  nodes: GraphNode[];
  edges: GraphEdge[];
}
